app/controllers/rooms_controller.rb

```diff
@@ -0,0 +1,69 @@
+class RoomsController < ApplicationController
+  before_action :set_room, only: [:edit, :update, :destroy, :archive]
+
+  def index
+    @rooms = Room.all
+  end
+
+  def show
+    @room = Room.find_by(slug: params[:slug])
+    raise ActiveRecord::RecordNotFound, "Couldn't find a room for that URL" unless @room
+    @topics = @room.topics.all
+  end
+
+  def new
+    @room = Room.new
+  end
+
+  def edit
+  end
+
+  def create
+    @room = Room.new(room_params)
+
+    respond_to do |format|
+      if @room.save
+        format.html { redirect_to @room, notice: 'Room was successfully created.' }
+        format.json { render :show, status: :created, location: @room }
+      else
+        format.html { render :new }
+        format.json { render json: @room.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def update
+    respond_to do |format|
+      if @room.update(room_params)
+        format.html { redirect_to root_url, notice: 'Room was successfully updated.' }
+        format.json { render :show, status: :ok, location: @room }
+      else
+        format.html { render :edit }
+        format.json { render json: @room.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def destroy
+    @room.destroy
+    respond_to do |format|
+      format.html { redirect_to root_url, notice: 'Room was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  def archive
+    topics = @room.topics.order('created_at DESC')
+    topics.where!('title ilike ?', "%#{params[:q]}%") if params[:q]
+    @grouped_topics = topics.group_by{ |t| t.created_at.to_date }
```

Comment:
I was originally going to do it exactly like you suggest!

But then the room's title ended up being "Awesome Room Archive". And I had gotten really enamored with the super clean user-facing URLs in this app. So a GET request to a URL of `/awesome/archive` seemed like it made sense!

From an end-user perspective, I think my approach does make sense—I'm GETting an archive, not PUTting something in an archive. But I can definitely see why this would be confusing, from a code readability perspective.

I just thought of maybe a simpler change that could keep the clean URL and communicate better with other code-readers. What if I rename it `archives` instead of `archive`?